gapii/client/capture.go

```diff
@@ -181,41 +186,109 @@ func (p *Process) Capture(ctx context.Context, start task.Signal, stop task.Sign
 	conn := p.conn
 	defer conn.Close()
 
-	var count siSize
-	started := false
+	if (p.Options.Flags & DeferStart) != 0 {
+		go func() {
+			if start.Wait(ctx) {
+				conn.Write([]byte{0x1, messageStartTrace})
```

Comment:
These now ignore the returned error from `conn.Write.`

Also, you should be able to do it in a single goroutine with a `select` (https://gobyexample.com/non-blocking-channel-operations)

Reply:
Have added error handling for conn.Write. By the way, all previous implementations (including current master) have ignored the write errors.

I would indeed be able to use a single goroutine with a select, but this would have to be in a loop to catch both signals, and would also need conditionals to filter out the fired signals. It is simpler and more readable with separate routines.